### PR TITLE
RuboCop

### PR DESCRIPTION
**Pull request by bquorning**
Start using RuboCop, and clean up the code a bit.

### CODE REVIEW

**Comment by bquorning** [comment]
cc @osheroff and @grosser

**Comment by grosser** [comment]
👍

File: lib/active_record_host_pool/pool_proxy.rb [hunk 1]
```diff
@@ -29,20 +31,12 @@ def __setobj__(spec)
...
-        _connection_proxy_for(real_connection, @config[:database])
-      rescue Exception => e
-        if rescuable_errors.any? { |r| e.is_a?(r) }
-          _connection_pools.delete(_pool_key)
-        end
-        Kernel.raise(e)
-      end
+      real_connection = _connection_pool.connection(*args)
+      _connection_proxy_for(real_connection, @config[:database])
+    rescue *rescuable_errors => e
```
Comment (bquorning) [inline lib/active_record_host_pool/pool_proxy.rb]: I’d like an extra pair of eyes on this particular diff. Yours, @osheroff ?

**Comment by bquorning** [comment]
cc @sandlerr and @grosser

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+# Offense count: 3
+Lint/RescueException:
+  Exclude:
+    - 'lib/active_record_host_pool/pool_proxy.rb'
+    - 'test/test_arhp_wrong_db.rb'
+
+# Offense count: 3
+Security/Eval:
+  Exclude:
+    - 'test/helper.rb'
```
Comment (grosser) [inline .rubocop_todo.yml]: should be fine to exclude eval in test folder

Comment (bquorning) [inline .rubocop_todo.yml]: Next PR.

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+inherit_from: .rubocop_todo.yml
+
+AllCops:
+  CacheRootDirectory: tmp
+  DisplayCopNames: true
+  Exclude:
+    - .git/**/*
+    - gemfiles/vendor/**/*
+    - vendor/**/*
+  TargetRubyVersion: 2.1
```
Comment (sandlerr) [inline .rubocop.yml]: if you change this to 2.3, the MutableConstant cops are happy

**Pull request merged**